ui/src/features/Canvas/components/NodeContextMenu/index.tsx

```diff
@@ -34,10 +34,18 @@ const NodeContextMenu: React.FC<Props> = ({
     onNodesChange?.([{ id, type: "remove" }]);
   }, [id, onNodesChange]);
 
-  const handleSecondaryNodeAction = useCallback(() => {
-    if (!id) return;
-    onSecondaryNodeAction?.(undefined, id, node.data.subworkflowId);
-  }, [id, node.data.subworkflowId, onSecondaryNodeAction]);
+  const handleSecondaryNodeAction = useCallback(
+    (allowNodeSettings?: boolean) => {
+      if (!id) return;
+      if (allowNodeSettings) {
+        onSecondaryNodeAction?.(undefined, id, undefined);
+        return;
+      }
+
+      onSecondaryNodeAction?.(undefined, id, node.data.subworkflowId);
+    },
+    [id, node.data.subworkflowId, onSecondaryNodeAction],
```

Comment:
Just keep in mind that we want to consolidate UX between nodes. Currently double click can do one of two things, which isn't necessarily great. Needs discussion